src/contracts/PreAMMBatcher.sol

```diff
@@ -404,4 +404,31 @@ contract PreAMMBatcher {
         }
         return newOrders;
     }
+
+    function isSorted(Order[] memory orders, bool reverse)
+        public
+        pure
+        returns (bool)
+    {
+        // Values used to traverse the list forwards or backwards depending on reverse.
+        // That is, reverse determines whether we want the orders to be ascendeing or descending
+        int256 start = 0;
+        int8 step = 1;
+        if (reverse) {
+            start = int256(orders.length) - 1;
+            step = -1;
+        }
```

Comment:
Wouldn't it be simpler to multiply `int256(orderA.buyAmount * orderB.sellAmount - orderB.buyAmount * orderA.sellAmount) * step > 0` to get the same effect? I find reasoning with the `start * i * step` a little hard to follow (I needed paper to check there were no off by 1 errors).

Reply:
Yes, This is a really nice suggestion! In that case, we wouldn't even need to traverse the list backwards, we can simply reverse the inequality!